framework/src/vectorpostprocessors/EigenvalueProstprocessor.C

```diff
@@ -0,0 +1,54 @@
+/****************************************************************/
+/*               DO NOT MODIFY THIS HEADER                      */
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*           (c) 2010 Battelle Energy Alliance, LLC             */
+/*                   ALL RIGHTS RESERVED                        */
+/*                                                              */
+/*          Prepared by Battelle Energy Alliance, LLC           */
+/*            Under Contract No. DE-AC07-05ID14517              */
+/*            With the U. S. Department of Energy               */
+/*                                                              */
+/*            See COPYRIGHT for full restrictions               */
+/****************************************************************/
+
+#include "EigenvaluePostprocessor.h"
+#include "libmesh/libmesh_config.h"
+
+template<>
+InputParameters validParams<EigenvaluePostprocessor>()
+{
+  InputParameters params = validParams<GeneralVectorPostprocessor>();
+  return params;
+}
+
+EigenvaluePostprocessor::EigenvaluePostprocessor(const InputParameters & parameters) :
+    GeneralVectorPostprocessor(parameters),
+    _eigen_values_real(declareVector("eigen_values_real")),
+    _eigen_values_imag(declareVector("eigen_values_imag")),
+    _nl_eigen(dynamic_cast<NonlinearEigenSystem &>(_fe_problem.getNonlinearSystemBase()))
+{}
+
+void
+EigenvaluePostprocessor::initialize()
+{}
+
+void
+EigenvaluePostprocessor::execute()
+{
+#if LIBMESH_HAVE_SLEPC
+  unsigned int n_converged_eigenvalues = _nl_eigen.getNumConvergedEigenvalues();
+  _eigen_values_real.clear();
+  _eigen_values_real.reserve(n_converged_eigenvalues);
+  _eigen_values_imag.clear();
+  _eigen_values_imag.reserve(n_converged_eigenvalues);
```

Comment:
Usually we clear in `initialize()` and resize in `execute()`. This doesn't matter much for GeneralUOs or GeneralPPs but it's a good pattern to follow. In this case however, I'd suggest that you drop the clear() altogether and opt for `resize`.